uefi/src/proto/shell_params.rs

```diff
@@ -0,0 +1,45 @@
+//! `ShellParams` protocol
+
+use crate::proto::unsafe_protocol;
+use crate::{data_types, Char16};
+use core::slice::from_raw_parts;
+use uefi_raw::protocol::shell_params::ShellParametersProtocol;
+
+#[cfg(feature = "alloc")]
+use crate::CStr16;
+#[cfg(feature = "alloc")]
+use alloc::string::String;
+#[cfg(feature = "alloc")]
+use alloc::string::ToString;
+
+/// The ShellParameters protocol.
+#[derive(Debug)]
+#[repr(transparent)]
+#[unsafe_protocol(ShellParametersProtocol::GUID)]
+pub struct ShellParameters(ShellParametersProtocol);
+
+impl ShellParameters {
+    /// Get an iterator of the shell parameter arguments
+    #[cfg(feature = "alloc")]
```

Comment:
Most of our APIs return `CStr16`s rather than converting to String. (Since there's a good chance that in real usage you'll pass these strings back into other uefi-rs methods, it makes sense to avoid the round-trip conversion from UCS-2 to UTF-8 and back). So this can change to:

```rust
pub fn args(&self) -> impl Iterator<Item = &CStr16> {
    self.get_args_slice()
        .iter()
        .map(|x| unsafe { CStr16::from_ptr(*x) })
}
```

And doesn't need to be gated behind `alloc` then.

Reply:
Ok sure, that's a good idea.